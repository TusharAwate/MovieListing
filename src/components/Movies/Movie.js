import React, { PureComponent } from 'react'
import './Movies.css'

class Movie extends PureComponent {
    constructor(props) {
        super(props)

        this.state = {
        }
    }

    render() {
        const {
            config,
            movie,
            genres,
        } = this.props

        const base_url = config && config.images.secure_base_url
        const genre_names = genres && genres.join(', ')


        return (
            <li className="Movie">
                {base_url && <picture>
                    <source srcSet={base_url + '/w92/' + movie.poster_path} media="(max-width: 500px)" />
                    <source srcSet={base_url + '/w185/' + movie.poster_path} />
                    <img className="Movie__Poster__Image" src={base_url + '/w185/' + movie.poster_path} alt={'Poster of ' + movie.title + ' movie'} />
                </picture>}
                <h2 className="Movie__Title">{movie.title}</h2>
                <h5 className="Movie__Genere">Genre: {genre_names}</h5>
            </li>
        )
    }
}

export default Movie