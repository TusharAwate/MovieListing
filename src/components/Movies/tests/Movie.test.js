import React from 'react';
import { render } from '@testing-library/react';
import Movie from '../Movie';
describe('Movie', () => {
    it('should render and match the snapshot', () => {
        const props = {
            movie: { poster_path: 'test_poster_path', title: 'test_title' },
            config: { images: { secure_base_url: 'test_secure_base_url' } },
            genres: ['genre1', 'genre2']
        };


        const {
            container: { firstChild },
        } = render(<Movie {...props} />);

        expect(firstChild).toMatchSnapshot();
    });

});

