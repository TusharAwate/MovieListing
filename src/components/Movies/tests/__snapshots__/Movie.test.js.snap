// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render and match the snapshot 1`] = `
<li
  class="Movie"
>
  <picture>
    <source
      srcset="test_secure_base_url/w92/test_poster_path"
      media="(max-width: 500px)"
    />
    <source
      srcset="test_secure_base_url/w185/test_poster_path"
    />
    <img
      class="Movie__Poster__Image"
      src="test_secure_base_url/w185/test_poster_path"
      alt="Poster of test_title movie"
    />
  </picture>
  <h2
    class="Movie__Title"
  >
    test_title
  </h2>
  <h5
    class="Movie__Genere"
  >
    Genre: 
    genre1, genre2
  </h5>
</li>
`;
